compiler/resolution/functionResolution.cpp

```diff
@@ -6752,9 +6752,20 @@ static void resolveNewSetupManaged(CallExpr* newExpr, Type*& manager) {
       if (ne->name == astr_chpl_manager) {
         manager = canonicalDecoratedClassType(ne->actual->typeInfo());
         expr->remove();
+        break;
       }
     }
   }
+
+  // adjust for the cases like 'new C()?'
+  if (manager == NULL) {
+    if (SymExpr* arg1 = toSymExpr(newExpr->get(1)))
+      if (DecoratedClassType* dt1 = toDecoratedClassType(arg1->symbol()->type))
+        if (dt1->getDecorator() == CLASS_TYPE_GENERIC_NILABLE)
```

Comment:
Why does this only check for GENERIC_NILABLE? It would make more sense to me if it checked for any CLASS_TYPE_GENERIC.

Reply:
In the case `new C()?` I am observing CLASS_TYPE_GENERIC_NILABLE. In what situations would other GENERIC variants occur?

Let me leave it this way for now. We will adjust once we decide what to do about this kind of expression on #13619.